server/superdesk/data_consistency/compare_repositories_tests.py

```diff
@@ -0,0 +1,128 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2013, 2014 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+import superdesk
+from eve.utils import ParsedRequest
+from superdesk.tests import TestCase
+from superdesk import get_resource_service
+from superdesk.data_consistency.compare_repositories import CompareRepositories
+from eve.methods.common import resolve_document_etag
+from time import sleep
+
+
+class CompareRepositoriesTestCase(TestCase):
+
+    def setUp(self):
+        super().setUp()
+        with self.app.app_context():
+            data = [{'headline': 'test {}'.format(i), 'slugline': 'rebuild {}'.format(i),
+                     'type': 'text' if (i % 2 == 0) else 'picture'} for i in range(1, 100)]
+            resolve_document_etag(data, 'archive')
+            superdesk.app.data._search_backend('archive').bulk_insert('archive', data)
+            get_resource_service('archive').post(data)
+            sleep(1)  # sleep so Elastic has time to refresh the indexes
+
+    def test_compare_repos(self):
+        with self.app.app_context():
```

Comment:
I agree that the repetition of the with statement was not necessary in most of the test cases here. Especially on those where I test the 'process_results' case. So maybe I can revisit this towards the end of this sprint. Thanks for the comment.